src/detectors/builtin/sendInLoop.ts

```diff
@@ -0,0 +1,184 @@
+import { CompilationUnit } from "../../internals/ir";
+import { foldStatements, isSelf } from "../../internals/tact";
+import { MistiTactWarning, Severity } from "../../internals/warnings";
+import { ASTDetector } from "../detector";
+import {
+  AstStatement,
+  AstExpression,
+  AstCondition,
+  AstStatementTry,
+  AstStatementTryCatch,
+  AstStatementWhile,
+  AstStatementRepeat,
+  AstStatementUntil,
+  AstStatementForEach,
+  idText,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+export class SendInLoop extends ASTDetector {
+  severity = Severity.HIGH;
+
+  async check(cu: CompilationUnit): Promise<MistiTactWarning[]> {
+    const warnings: MistiTactWarning[] = [];
+    this.collectWarnings(cu, warnings);
+    return warnings;
+  }
+
+  private collectWarnings(
+    cu: CompilationUnit,
+    warnings: MistiTactWarning[],
+  ): void {
+    const processedLoopIds = new Set<number>();
+    for (const node of cu.ast.getProgramEntries()) {
+      foldStatements(
+        node,
+        (accumulatedWarnings, statement) => {
+          this.analyzeStatement(
+            statement,
+            accumulatedWarnings,
+            false,
+            processedLoopIds,
+          );
+          return accumulatedWarnings;
+        },
+        warnings,
+        { flatStmts: true },
+      );
+    }
+  }
+
+  private isSendCall(expr: AstExpression): boolean {
+    const staticSendFunctions = ["send", "nativeSendMessage"];
+    const selfMethodSendFunctions = ["reply", "forward", "notify", "emit"];
+
+    return (
+      (expr.kind === "static_call" &&
+        staticSendFunctions.includes(idText(expr.function))) ||
+      (expr.kind === "method_call" &&
+        isSelf(expr.self) &&
+        selfMethodSendFunctions.includes(idText(expr.method)))
+    );
+  }
+
+  private analyzeStatement(
+    statement: AstStatement,
+    warnings: MistiTactWarning[],
+    inLoop: boolean,
+    processedLoopIds: Set<number>,
+  ): void {
+    // check If the statement is a loop!
+    if (this.isLoopStatement(statement)) {
+      // Avoid processing the same loop multiple times
+      if (!processedLoopIds.has(statement.id)) {
+        processedLoopIds.add(statement.id);
+        inLoop = true;
+      }
+    }
+
+    // If stmt is an expr statement we are inside loop
```

Comment:
I didn't get it. The comments seems to be misleading, but the logic seems correct.